api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationTranslatorImpl.java

```diff
@@ -153,40 +208,14 @@ public Observation toFhirResource(@Nonnull Obs observation) {
 		return obs;
 	}
 	
-	@Override
-	public Obs toOpenmrsType(@Nonnull Observation fhirObservation) {
-		notNull(fhirObservation, "The Observation object should not be null");
-		return toOpenmrsType(new Obs(), fhirObservation);
-	}
-	
-	@Override
-	public Obs toOpenmrsType(Obs existingObs, Observation observation, Supplier<Obs> groupedObsFactory) {
-		notNull(existingObs, "The existing Obs object should not be null");
-		notNull(observation, "The Observation object should not be null");
-		
-		observationStatusTranslator.toOpenmrsType(existingObs, observation.getStatus());
-		
-		existingObs.setEncounter(encounterReferenceTranslator.toOpenmrsType(observation.getEncounter()));
-		existingObs.setPerson(patientReferenceTranslator.toOpenmrsType(observation.getSubject()));
+	private class ObservationTranslatorContext {
```

Comment:
It might not be necessary if you just annotate 
```
@Cacheable
CodeableConcept toFhirResource(@Nonnull Concept concept) {}
```
Please see my comment above.